src/Microsoft.Azure.EventHubs.Processor/Lease.cs

```diff
@@ -35,15 +34,15 @@ protected Lease(Lease source)
 
         public long Epoch { get; set; }
 
-        public virtual bool IsExpired() 
+        public virtual Task<bool> IsExpired() 
         {
     	    // this function is meaningless in the base class
-    	    return false;
+            return Task.FromResult(false);
         }
 
-        public virtual string GetStateDebug()
+        public virtual Task<string> GetStateDebug()
```

Comment:
@serkantkaraca would it add value to decorate with [`DebuggerDisplayAttribute`](https://msdn.microsoft.com/en-us/library/ms228992(v=vs.110).aspx)? I see the value in this, so perhaps `DebuggerTypeProxy` could do the job and leave public API cleaner?